fs/config/config.go

```diff
@@ -229,6 +229,14 @@ func SetConfigPath(path string) (err error) {
 	if path == "" || path == os.DevNull {
 		configPath = ""
 	} else {
+		if runtime.GOOS == "windows" {
+			// On Windows prevent referring to legacy device, which will happen if basename without extension
+			// is one of the reserved names (https://docs.microsoft.com/en-gb/windows/win32/fileio/naming-a-file)
+			if reserved, _ := regexp.MatchString(`^(?i:con|prn|aux|nul|com[1-9]|lpt[1-9])(?:\.|$)`, filepath.Base(path)); reserved {
```

Comment:
Ewww!  You could put this into `lib/file` as `IsReserved` or something like that?

I note there is an implementation of this in the standard library but it isn't exported :-(

```go
/ reservedNames lists reserved Windows names. Search for PRN in
// https://docs.microsoft.com/en-us/windows/desktop/fileio/naming-a-file
// for details.
var reservedNames = []string{
        "CON", "PRN", "AUX", "NUL",
        "COM1", "COM2", "COM3", "COM4", "COM5", "COM6", "COM7", "COM8", "COM9",
        "LPT1", "LPT2", "LPT3", "LPT4", "LPT5", "LPT6", "LPT7", "LPT8", "LPT9",
}

// isReservedName returns true, if path is Windows reserved name.
// See reservedNames for the full list.
func isReservedName(path string) bool {
        if len(path) == 0 {
                return false
        }
        for _, reserved := range reservedNames {
                if strings.EqualFold(path, reserved) {
                        return true
                }
        }
        return false
}
```

It also seems to be wrong as it doesn't consider `aux.c` to be a reserved name, whereas Windows certainly does (I can tell you this from bitter personal experience porting a large C++ code base to Windows!).

Reply:
> Ewww! You could put this into lib/file as IsReserved or something like that?

Yes, that will be better.

> I can tell you this from bitter personal experience porting a large C++ code base to Windows!

My condolences!